sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Core/ConnectionStringProperties.cs

```diff
@@ -84,12 +94,23 @@ internal struct ConnectionStringProperties
                 throw new ArgumentException(Resources.OnlyOneEventHubNameMayBeSpecified, connectionStringArgumentName);
             }
 
+            // The connection string may contain a precomputed shared access signature OR a shared key name and value,
+            // but not both.
+
+            if ((!string.IsNullOrEmpty(SharedAccessSignature))
+                && ((!string.IsNullOrEmpty(SharedAccessKeyName)) || (!string.IsNullOrEmpty(SharedAccessKey))))
+            {
+                throw new ArgumentException(Resources.OnlyOneSharedAccessAuthorizationMayBeSpecified, connectionStringArgumentName);
+            }
```

Comment:
Would it be simpler if we added a new ctor that accepted only `sharedAccessSignature` and not `sharedAccessKey` / `sharedAccessKeyName` to prevent the invalid combo?

Reply:
Good thought, but that changes some of the chain of responsibility.  That would cause parsing the connection string to fail, which it should only do if the string is malformed.   The purpose of this method is to check the set of tokens in the string, once fully parsed, and determine the validity of the set.